lib/fluent/plugin/parser_syslog.rb

```diff
@@ -137,6 +153,91 @@ def parse_plain(text, &block)
 
         yield time, record
       end
+
+      SPLIT_CHAR = ' '.freeze
+      PRI_START_CHAR = '<'.freeze
+
+      def parse_rfc3164(text, &block)
+        pri = nil
+        start = 0
+        if @with_priority
+          if text.start_with?(PRI_START_CHAR)
+            i = text.index('>'.freeze, 1)
+            pri = text.slice(1, i - 1).to_i
+            start = i + 1
+          else
+            yield nil, nil
+            return
+          end
+        end
+
+        # header part
+        diff = 15 # skip Mmm dd hh:mm:ss
+        time_end = text[start + diff]
+        if time_end == SPLIT_CHAR
+          time_str = text.slice(start, diff)
+          start += 16 # time + ' '
+        elsif time_end == '.'.freeze
+          # support subsecond time
+          i = text.index(SPLIT_CHAR, diff)
+          time_str = text.slice(start, i - start)
+          start = i + 1
+        else
+          yield nil, nil
+          return
+        end
+
+        i = text.index(SPLIT_CHAR, start)
+        if i.nil?
+          yield nil, nil
+          return
+        end
+        diff = i - start
+        host = text.slice(start, diff)
+        start += (diff + 1)
+
+        i = text.index(SPLIT_CHAR, start)
```

Comment:
If the text is `${TIMESTAMP} ${HOST} test:val`, this parser won't work. and I think `test:val` is valid syslog message.
`ident` should be `test` and `msg` should be `val`

Reply:
This is very difficult problem because rfc3164 doesn't define how ident and message are separated. For the experience, here is an actual examples:

```
HEADER ident msg  # No problem
HEADER ident: msg # No problem
HEADER ident:msg  # Existing products send this as only msg, not ident:msg.
```

So to support both cases, adding option is better.